src/audio/component.c

```diff
@@ -505,3 +505,51 @@ int comp_copy(struct comp_dev *dev)
 
 	return ret;
 }
+
+#if CONFIG_IPC_MAJOR_4
+static uint32_t get_sample_group_size_in_bytes(const struct ipc4_audio_format fmt)
+{
+	return (fmt.depth >> 3) * fmt.channels_count;
+}
+
+static uint32_t get_one_ms_in_bytes(const struct ipc4_audio_format fmt)
+{
+#ifdef ROUND_UP
+	/* TODO Reference Firmware also has systick multiplier and divider in this equation */
+	return get_sample_group_size_in_bytes(fmt) *
+			(ROUND_UP(fmt.sampling_frequency, 1000) / 1000);
+#else
+	return get_sample_group_size_in_bytes(fmt) *
+				(((fmt.sampling_frequency + 999) / 1000));
+#endif
+}
+#endif
+
+void comp_update_ibs_obs_cpc(struct comp_dev *dev)
+{
+#if CONFIG_IPC_MAJOR_4
+	int ret;
+	struct ipc4_base_module_cfg source_src_cfg;
```

Comment:
why is this called `source_src_cfg`? What source and what src is it referring to?